open_academy/tests/test_course.py

```diff
@@ -0,0 +1,44 @@
+from psycopg2.errors import CheckViolation
+from psycopg2.errors import UniqueViolation
+
+from odoo.tests.common import TransactionCase
+from odoo.tools import mute_logger
+
+
+class GlobalTestCourse(TransactionCase):
+    """Course model test"""
+    def setUp(self):
+        super().setUp()
+        self.course = self.env['course']
+
+    def create_course(self, title, description):
+        """Global function to create courses"""
+        course_id = self.course.create({
+            'title': title,
+            'description': description,
+        })
+        return course_id
+
+    @mute_logger('odoo.sql_db')
+    def test_01_course_same_name_description(self):
+        """Case when course's title is same as course's description: Error sql constraint"""
+        with self.assertRaisesRegex(
+            CheckViolation,
+            'new row for relation "course" violates check constraint "course_check_title_description'
+        ):
+            self.create_course('test', 'test')
+
+    @mute_logger('odoo.sql_db')
+    def test_02_two_courses_same_name(self):
+        """Case when the user try to create repeated courses' title: Error sql constraint"""
+        with self.assertRaisesRegex(
+            UniqueViolation,
+            'duplicate key value violates unique constraint "course_title_unique"'
+        ):
+            self.create_course('test', 'test_description_01')
+            self.create_course('test', 'test_description_02')
+
+    def test_03_duplicate_course(self):
```

Comment:
why do you need to test this? or if you are checking the data is copied as expected, you could check that maybe using `self.assertRecordValues` (https://www.odoo.com/documentation/15.0/developer/howtos/rdtraining/E_unittest.html)

Reply:
I just was following the Moy's video tutorial, but if this is better, I will change it.